src/Controllers/TicketsController.php

```diff
@@ -68,15 +68,18 @@ public function show (EnabledAgentsRepository $agents, TicketsRepository $ticket
         $ticket = $tickets->with($this->showRelations)->findOrFail($id);
 
         if ($ticket->teamAssignment) {
-            $agentsCollection = auth()->user()->is_super
-            ? $agents->clone()->get()
-            : $agents->clone()->inTeam($ticket->teamAssignment->team)->get();
-        } elseif (auth()->user()->is_super) {
-            $agentsCollection = $agents->clone()->get();
+            if (isset(auth()->user()->agent)) {
+                $agentsCollection = auth()->user()->agent->is_super
+                ? $agents->clone()->get()
+                : $agents->clone()->inTeam($ticket->teamAssignment->team)->get();
+            } else {
+                $agentsCollection = $agents->clone()->inTeam($ticket->teamAssignment->team)->get();
+            }
         } elseif ($ticket->assignment) {
             foreach ($ticket->assignment->assignee->teamLeads as $key => $team) {
```

Comment:
This entire thing is ripe for a refactor at some point, this much cyclomatic complexity in a controller is a sign that an abstraction is required.